ligo/stablecoin/actions.ligo

```diff
@@ -144,7 +144,9 @@ function debit_from
       then failwith ("FA2_INSUFFICIENT_BALANCE")
       else
       { const updated_balance : nat = abs (curr_balance - parameter.amount)
-      ; updated_ledger := Big_map.update(parameter.from_, Some (updated_balance), updated_ledger)
+      ; if updated_balance = 0n
+          then updated_ledger := Big_map.remove(parameter.from_, updated_ledger);
+          else updated_ledger := Big_map.update(parameter.from_, Some (updated_balance), updated_ledger);
```

Comment:
@gromakovsky I have refactored that section a bit as you suggested.

Reply:
Did you check that the contract became smaller? There were some cases when I expect a certain change to make the contract smaller, but it became larger :) So in this case I am not sure whether my suggested change is good or not. It may be good, that's why I suggested it, but it may be bad as well.

Reply:
@gromakovsky Did a simple lines of code comparison on generated contract now..

The contract size now is 1597 lines. In `master` it was 1600. And in this branch before refactoring, it was 1613 lines.

Reply:
Now that we have `stablecoin-client` it should be quite easy to originate the contract to carthagenet and see operation costs in block explorer. Can you do that please (for latest version and before refactoring)? You can download it from CI, then you won't need to build anything. Should take 5 minutes unless I am missing something.

Reply:
Ah yes. Il do.

Reply:
@gromakovsky Took more time that expected because I mixed up builds and ended up with same contract for different branches...anyway, here is the block explorer links..

Latest: https://carthage.tzkt.io/ooDPJ8nGyxNRS6egmSY9robVKSQ9Xw6srwGeQj5aMAHKsTzNags
Before refactoring : https://carthage.tzkt.io/op8y4mNhM3CJe8S4DYnnsr3YSnydXrgB2XWBRzzKNEM47SgEF6L
Master: https://carthage.tzkt.io/oorYmjnU33oyjwRvDedKdGM6hNh6U3cQs67f6EnW76NysQVz3TL

Gas costs:

Latest: 293945
Before refactor: 296763
Master : 293739

Reply:
Ok, cool, thanks.